### DISCUSSION TITLE
How do I express one variable with respect to others?

### DISCUSSION THREAD

**Discussion by ritog** [sympy/sympy#24049] [Q&A] [answered]
I have an equation that is very large to be manipulated by hand. There are many variables.

My goal is to express one variable with respect to the others.

If I have an equation like this:

$\frac{a}{b} = \frac{a+b}{a^2 + 2b -3}$

How do I express $a$ with respect to $b$?

Going through the help available on the web, I have found that the `solve()` method works when there are numerical values involved, and not with expressing one variable with the other variables.

How do I solve this?

**Comment by smichr** [comment, marked as answer]
`eq = Eq(x/y,(x+y)/(x**2+2*y+3)); solve(eq, x) -> 3 solutions for x in terms of y` and `solve(eq, y) -> 2 solutions for y in terms of x`. Is that what you are looking for?

**Reply by ritog** [reply]
Not quite.

But this is indeed close to the answer I described.

I am going to mark this as answer.

I have found my solution elsewhere. Thanks anyway.

Sorry for keeping this post as unanswered for so long.

**Reply by evbernardes** [reply]
> I have found my solution elsewhere. Thanks anyway.


It would be nice if you put the answer you found as a comment here, if someone else ever has the same problem.
